backend/oasst_backend/prompt_repository.py

```diff
@@ -488,3 +496,178 @@ def close_task(self, post_id: str, allow_personal_tasks: bool = False):
+        end_date: Optional[datetime.datetime] = None,
+        only_roots: bool = False,
+        deleted: Optional[bool] = None,
+    ) -> list[Post]:
+        if not self.api_client.trusted and not api_client_id:
+            # Let unprivileged api clients query their own messages without api_client_id being set
+            api_client_id = self.api_client.id
+
+        if not self.api_client.trusted and api_client_id != self.api_client.id:
+            # Unprivileged api client asks for foreign messages
+            raise OasstError("Forbidden", OasstErrorCode.API_CLIENT_NOT_AUTHORIZED, HTTP_403_FORBIDDEN)
+
+        messages = self.db.query(Post)
+        if user_id:
+            messages = messages.filter(Post.person_id == user_id)
+        if username:
+            messages = messages.join(Person)
+            messages = messages.filter(Person.username == username)
+        if api_client_id:
+            messages = messages.filter(Post.api_client_id == api_client_id)
+
+        if start_date:
+            messages = messages.filter(Post.created_date >= start_date)
+        if end_date:
+            messages = messages.filter(Post.created_date < end_date)
+
+        if only_roots:
+            messages = messages.filter(Post.parent_id.is_(None))
+
+        if deleted is not None:
+            messages = messages.filter(Post.deleted == deleted)
+
+        if desc:
+            messages = messages.order_by(Post.created_date.desc())
+        else:
+            messages = messages.order_by(Post.created_date.asc())
+
+        if limit is not None:
+            messages = messages.limit(limit)
+
+        # TODO: Pagination could be great at some point
+        return messages.all()
+
+    def mark_messages_deleted(self, messages: Post | UUID | list[Post | UUID], recursive: bool = True):
+        if isinstance(messages, (Post, UUID)):
+            messages = [messages]
+
+        ids = []
+        for message in messages:
+            if isinstance(message, UUID):
+                ids.append(message)
+            elif isinstance(message, Post):
+                ids.append(message.id)
+            else:
+                raise OasstError("Server error", OasstErrorCode.SERVER_ERROR, HTTPStatus.INTERNAL_SERVER_ERROR)
+
+        query = update(Post).where(Post.id.in_(ids)).values(deleted=True)
+        self.db.execute(query)
+
+        parent_ids = ids
+        if recursive:
+            while parent_ids:
+                query = update(Post).filter(Post.parent_id.in_(parent_ids)).values(deleted=True).returning(Post.id)
+
+                parent_ids = self.db.execute(query).scalars().all()
+
+        self.db.commit()
+
+    def get_stats(self):
+        deleted = self.db.query(Post.deleted, func.count()).group_by(Post.deleted)
+        nthreads = self.db.query(None, func.count(Post.id)).filter(Post.parent_id.is_(None))
+        query = deleted.union_all(nthreads)
+        result = {k: v for k, v in query.all()}
+        return {
```

Comment:
- maybe use here the `SystemStats` class from protocol instead dict?
- return type annotation for get_stats() is missing

Reply:
Changed return type and value